utils.py

```diff
@@ -352,40 +349,58 @@ def print_attributes(vds, path=None):
     """
-    Get all 1-, A-, G- numbered annotations from a VDS based on the Number va attributes.
-    In addition returns arrays with no Number or Number=. va attribute separately
-    :param vds: Input VDS
-    :param root: Place to find annotations (defaults to va.info)
-    :return: annotations, a_annotations, g_annotations, dot_annotations as list[Field]
+        Get numbered annotations from a VDS variant schema based on their `Number` va attributes.
+    The numbered annotations are returned as a dict with the Number as the key and a list of tuples (field_path, field) as values.
+    All annotations that do not have a Number attribute are returned under the key `None`
+    :param TStruct schema: Input variant schema
+    :param str root: Root path to get annotations (defaults to va)
+    :param bool recursive: Whether to go recursively to look for Numbered annotations in TStruct fields
+    :return: Dictionary containing annotations
+    :rtype: dict of tuple(str, Field)
     """
-    a_annotations = []
-    g_annotations = []
-    dot_annotations = []
-    annotations = []
+    annotations = group_annotations_by_attribute(schema, 'Number', root, recursive)
+    logger.info("Found the following fields:")
+    for k, v in annotations.iteritems():
+        if k is not None:
+            logger.info("{}-based annotations: {}".format(k, ",".join([fields[0] for fields in v])))
+        else:
+            logger.info("Annotations with no number: {}".format(",".join([fields[0] for fields in v])))
+
+    return annotations
+
+def group_annotations_by_attribute(schema, grouping_key , root, recursive = False):
+    """
+    Groups annotations in a dictionnary by the given attribute key.
+    All annotations that do not have a Number attribute are returned under the key `None`
+
+    :param TStruct schema: Input schema
+    :param str root: Root path to get annotations
+    :param bool recursive: Whether to go recursively to look for Numbered annotations in TStruct fields
+    :return: Dictionary containing annotations
+    :rtype: dict of tuple(str, Field)
+    """
+    annotations = defaultdict(list)
+
+    if '.' in root:
+        fields = get_ann_type(root, schema)
+    else:
+        fields = schema
 
-    release_info = get_ann_type(root, vds.variant_schema)
-    for field in release_info.fields:
+    for field in fields.fields:
+        path = '{}.{}'.format(root, field.name)
         if isinstance(field.typ, TArray):
-            if 'Number' in field.attributes:
-                number = field.attributes['Number']
-                if number == "A":
-                    a_annotations.append(field)
-                elif number == "G":
-                    g_annotations.append(field)
-                else:
-                    dot_annotations.append(field)
+            if grouping_key in field.attributes:
+                annotations[field.attributes[grouping_key]].append((path, field))
```

Comment:
I'm not wild about the tuple of 2 as a value. It's harder to use and you have to remember which is which. Is this unavoidable? (It may be, but just thinking it through)

Reply:
The issue here is that the `Field` `name` doesn't have the path. How about using a namedtuple then?

Reply:
Yeah that works